File: src/main/java/com/otaviolarrosa/identitymanager/infrastructure/database/repositories/UserRepository.java
package com.otaviolarrosa.identitymanager.infrastructure.database.repositories;

import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

import com.otaviolarrosa.identitymanager.infrastructure.database.entities.UserEntity;

public interface UserRepository extends JpaRepository<UserEntity, Long> { 
	UserEntity findByUserCode(UUID userCode);
	UserEntity findByEmail(String email);
}

File: src/main/kotlin/com/otaviolarrosa/singleSignOn/infrastructure/database/repositories/UserRepository.kt
package com.otaviolarrosa.singleSignOn.infrastructure.database.repositories

import com.otaviolarrosa.singleSignOn.infrastructure.database.entities.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<UserEntity?, Long?> {
    fun findByUserCode(userCode: UUID?): UserEntity
    fun findByEmail(email: String?): UserEntity?
}
